src/components/h5p-button.js

```diff
@@ -56,8 +56,34 @@ H5P.Components.Button = (function () {
       H5P.Tooltip(button, { text: params.tooltip, position: params.tooltipPosition ?? 'top' });
     }
 
+    if (params.icon && H5P.Components.ButtonIconOnlyObserver) {
```

Comment:
I think the ownership would be more clear if you used `H5P.Components.Button.IconOnlyObserver` instead. Otherwise I'd put the resizer code into a separate file along with the other components.